A102-xds-grpc-service.md

```diff
@@ -0,0 +1,305 @@
+  type that it does not support.  Note that unlike channel credentials,
+  call credentials are optional, and there can be more than one, so the
+  client will need to iterate over the entire list, but it's valid if
+  none of the specified types are supported.  If the client finds a
+  supported credential type but the config is invalid, then the xDS
+  resource will be NACKed.  The following extensions will be supported
+  in this field:
+
+  - `envoy.extensions.grpc_service.call_credentials.access_token.v3.AccessTokenCredentials`:
+    In this message:
+    - `token`: Required.  The access token.  The token will be added as
+      an `authorization` header with header `Bearer ` (note trailing
+      space) followed by the value of this field.  Note that the
+      token will not be sent on the wire unless the connection has
+      security level PRIVACY_AND_INTEGRITY.
+
+Note that this will require extending the channel credentials and call
+credentials registries to support configuration via these protos, in
+additional to the JSON formats that they already support for the gRPC
+xDS bootstrap config.
+
+### `GrpcService` Proto Validation
+
+When validating a `GrpcService` proto, the following fields will be used:
+- [`google_grpc`](https://github.com/envoyproxy/envoy/blob/7ebdf6da0a49240778fd6fed42670157fde371db/api/envoy/config/core/v3/grpc_service.proto#L303):
+  This field must be set.  Inside of it:
+  - [`target_uri`](https://github.com/envoyproxy/envoy/blob/7ebdf6da0a49240778fd6fed42670157fde371db/api/envoy/config/core/v3/grpc_service.proto#L254):
+    Must be set to a valid gRPC target URI.  The target URI must be
+    checked against the resolver registry during xDS resource
+    validation.
+  - `channel_credentials_plugin`: See above.
+  - `call_credentials_plugin`: See above.
+  - `channel_credentials`, `call_credentials`,
+    `credentials_factory_name`, and `config`: Ignored; we will use the
+    new credential plugin fields above instead.
+  - `stat_prefix`: Ignored; not relevant to gRPC.
+  - `per_stream_buffer_limit_bytes`: Ignored.  We don't have a use-case
+    for this right now but could add it later if needed.
+  - `channel_args`: Ignored.  Not supportable across languages in gRPC.
+- [`timeout`](https://github.com/envoyproxy/envoy/blob/7ebdf6da0a49240778fd6fed42670157fde371db/api/envoy/config/core/v3/grpc_service.proto#L308):
+  If set, this will be used to set the deadline on RPCs sent to the
+  side-channel service.  The value must obey the restrictions specified in
+  the [`google.protobuf.Duration`
+  documentation](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration),
+  and it must have a positive value.
+- [`initial_metadata`](https://github.com/envoyproxy/envoy/blob/7ebdf6da0a49240778fd6fed42670157fde371db/api/envoy/config/core/v3/grpc_service.proto#L315):
+  If present, specifies headers to be added to RPCs sent to the side-channel
```

Comment:
Did we talk to security team about this feature? I thought setting header configuration in general was restricted to trusted sources.